pkg/kubeapiserver/authenticator/patch_authenticator.go

```diff
@@ -54,18 +54,22 @@ func AddOAuthServerAuthenticatorIfNeeded(tokenAuthenticators []authenticator.Tok
 		panic(err)
 	}
 
-	// add our oauth token validator
-	validators := []oauth.OAuthTokenValidator{oauth.NewExpirationValidator(), oauth.NewUIDValidator()}
+	// Tokens are valid for their lifetime unless this value is overridden.
+	tokenTimeout := int32(0)
 	if enablement.OpenshiftConfig().OAuthConfig != nil {
-		if inactivityTimeout := enablement.OpenshiftConfig().OAuthConfig.TokenConfig.AccessTokenInactivityTimeoutSeconds; inactivityTimeout != nil {
-			timeoutValidator := oauth.NewTimeoutValidator(oauthClient.OauthV1().OAuthAccessTokens(), oauthInformer.Oauth().V1().OAuthClients().Lister(), *inactivityTimeout, oauthvalidation.MinimumInactivityTimeoutSeconds)
-			validators = append(validators, timeoutValidator)
-			enablement.AddPostStartHookOrDie("openshift.io-TokenTimeoutUpdater", func(context genericapiserver.PostStartHookContext) error {
-				go timeoutValidator.Run(context.StopCh)
-				return nil
-			})
+		if inactivityTimeout := enablement.OpenshiftConfig().OAuthConfig.TokenConfig.AccessTokenInactivityTimeout; inactivityTimeout != nil {
+			tokenTimeout = int32(inactivityTimeout.Seconds())
 		}
 	}
+
+	timeoutValidator := oauth.NewTimeoutValidator(oauthClient.OauthV1().OAuthAccessTokens(), oauthInformer.Oauth().V1().OAuthClients().Lister(), tokenTimeout, oauthvalidation.MinimumInactivityTimeoutSeconds)
+	// add our oauth token validator
+	validators := []oauth.OAuthTokenValidator{oauth.NewExpirationValidator(), oauth.NewUIDValidator(), timeoutValidator}
+	enablement.AddPostStartHookOrDie("openshift.io-TokenTimeoutUpdater", func(context genericapiserver.PostStartHookContext) error {
+		go timeoutValidator.Run(context.StopCh)
+		return nil
+	})
```

Comment:
should we move this inside the `if enablement.OpenshiftConfig().OAuthConfig != nil` ?

Reply:
what would be the point, we mean to run it all the time, no?

Reply:
In the oauth-off mode we will get (keycloak), don't we want to disable this?

Reply:
these authenticators are only configured if webhook authenticators aren't, so that's ok

Reply:
see https://github.com/openshift/kubernetes/blob/f325571df89d6ccc7f501a06a19be7b5d9bad743/pkg/kubeapiserver/authenticator/config.go#L185-L199